pkg/reconciler/cluster/syncer/syncer_controller.go

```diff
@@ -77,51 +77,7 @@ func NewController(
 	}, nil
 }
 
-type preparedController struct {
-	Controller
-}
-
-type PreparedController struct {
-	*preparedController
-}
-
-func (c *Controller) Prepare() (PreparedController, error) {
-	// TODO(ncdc): does this need a per-cluster client?
-	// TODO(sttts): make resilient to errors
-	discoveryClient := c.apiExtensionsClient.Discovery()
-	serverGroups, err := discoveryClient.ServerGroups()
-	if err != nil {
-		return PreparedController{}, err
-	}
-	for _, apiGroup := range serverGroups.Groups {
-		if genericcontrolplanescheme.Scheme.IsGroupRegistered(apiGroup.Name) {
-			for _, version := range apiGroup.Versions {
-				gv := schema.GroupVersion{
-					Group:   apiGroup.Name,
-					Version: version.Version,
-				}
-				if genericcontrolplanescheme.Scheme.IsVersionRegistered(gv) {
-					apiResourceList, err := discoveryClient.ServerResourcesForGroupVersion(gv.String())
-					if err != nil {
-						return PreparedController{}, err
-					}
-					for _, apiResource := range apiResourceList.APIResources {
-						gvk := gv.WithKind(apiResource.Kind)
-						if !strings.Contains(apiResource.Name, "/") && genericcontrolplanescheme.Scheme.Recognizes(gvk) {
-							c.syncerManager.genericControlPlaneResources = append(c.syncerManager.genericControlPlaneResources, gv.WithResource(apiResource.Name))
```

Comment:
This code was to also include generic control plane resources, like config maps or secrets, in what can be synced by the syncer, even though there was no related API imports, since those resource types are by default present in every KCP workspace.

Reply:
We should not use discovery for this. If this was about configmaps and secrets, we have to hardcode those very explicitly in the syncer. We cannot use some implicit mechanism like asking discovery.

Reply:
Completely agree that the change made in commit https://github.com/kcp-dev/kcp/pull/619/commits/4bb843b2dc004344f7d0993223901cdf7d9fb11c is sufficient and better when hard-coded.